### ISSUE TITLE
Empty table cells aren't meaningfully announced by screen readers

### ISSUE THREAD

**Issue by coforma-jamie** [department-of-veterans-affairs/vets-design-system-documentation#4841]
# Bug Report

- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

## What happened

By default, empty cells in `<va-table>` add `--` via CSS `content`.

`--` isn't announced meaningfully to screen reader users. Screen readers will either announce "dash dash" or not announce anything at all.

https://github.com/user-attachments/assets/287a49dd-b809-441b-a600-95064563fc2b

There are two avenues to take that can fix this:
1. Change `--` to something meaningful, like `blank` or `empty`
2. Add alternative text to `content`, like this: `content: '---'/'blank'`

Note that while the `content` solution technically works, there are known issues with it: it doesn't auto-translate, it's not available if the user turns off styles, and it's not searchable / selectable. ([Article with more information](https://www.sarasoueidan.com/blog/alt-text-for-css-generated-content/))

Because of these issues, we should also add VADS guidance instructing teams to not create empty cells in the first place, and give examples of "empty" or "not applicable" phrases to use instead.

## What I expected to happen

All table cells' content are announced by assistive technology

## Reproducing

- CSS Library version:
- Device: Macbook / VoiceOver
- Browser: Chrome

**URL(s) where this bug occurs**
https://design.va.gov/storybook/?path=/story/uswds-va-table--with-missing-data

**Steps to reproduce:**

1. Turn on screen reader
2. Navigate to the empty (`--`) table cell

## Urgency

How urgent is this request? Please select the appropriate option below and/or provide details

- [ ] This bug is blocking work currently in progress
- [x] This bug is affecting work currently in progress but we have a workaround
- [ ] This bug is blocking work planned within the next few sprints
- [ ] This bug is not blocking any work
- [ ] Other

**Comment by coforma-jamie** [comment]
Great find @josephrlee!

Long story short, leaving a cell blank or putting in `--` leads to non-descriptive screen reader announcements. Some may say "blank," others "dash dash," others don't say anything at all. To avoid this problem, we want to put something meaningful in every table cell, even if that meaning is "not applicable" or "empty."
